src/Symfony/Component/Validator/Validation.php

```diff
@@ -39,13 +61,14 @@ public static function createCallable($constraintOrValidator = null, Constraint
 
         $validator = $validator ?? self::createValidator();
 
-        return static function ($value) use ($constraints, $validator) {
-            $violations = $validator->validate($value, $constraints);
-            if (0 !== $violations->count()) {
-                throw new ValidationFailedException($value, $violations);
+        return static function ($value, ConstraintViolationListInterface $violations = null) use ($constraints, $validator) {
+            $valueViolations = $validator->validate($value, $constraints);
```

Comment:
why not set $violations by ref?

Reply:
Fixed, I've almost never used by ref arguments, so didn't know this was possible

Reply:
psalm fixed it :} https://psalm.dev/docs/annotating_code/supported_annotations/#param-out-psalm-param-out